public/main.js

```diff
@@ -0,0 +1,350 @@
+      this.gameState[8] === "X"
+    ) {
+      setTimeout(function () {
+        alert("Player X has won!");
+      }, 0);
+      setTimeout(function () {
+        location.reload();
+      }, 0);
+    } else if (
+      this.gameState[2] === "X" &&
+      this.gameState[4] === "X" &&
+      this.gameState[6] === "X"
+    ) {
+      setTimeout(function () {
+        alert("Player X has won!");
+      }, 0);
+      setTimeout(function () {
+        location.reload();
+      }, 0);
+    } else if (
+      this.gameState[0] === "O" &&
+      this.gameState[1] === "O" &&
+      this.gameState[2] === "O"
+    ) {
+      setTimeout(function () {
+        alert("Player O has won!");
+      }, 0);
+      setTimeout(function () {
+        location.reload();
+      }, 0);
+    } else if (
+      this.gameState[3] === "O" &&
+      this.gameState[4] === "O" &&
+      this.gameState[5] === "O"
+    ) {
+      setTimeout(function () {
+        alert("Player O has won!");
+      }, 0);
+      setTimeout(function () {
+        location.reload();
+      }, 0);
+    } else if (
+      this.gameState[6] === "O" &&
+      this.gameState[7] === "O" &&
+      this.gameState[8] === "O"
+    ) {
+      setTimeout(function () {
+        alert("Player O has won!");
+      }, 0);
+      setTimeout(function () {
+        location.reload();
+      }, 0);
+    } else if (
+      this.gameState[0] === "O" &&
+      this.gameState[3] === "O" &&
+      this.gameState[6] === "O"
+    ) {
+      setTimeout(function () {
+        alert("Player O has won!");
+      }, 0);
+      setTimeout(function () {
+        location.reload();
+      }, 0);
+    } else if (
+      this.gameState[1] === "O" &&
+      this.gameState[4] === "O" &&
+      this.gameState[7] === "O"
+    ) {
+      setTimeout(function () {
+        alert("Player O has won!");
+      }, 0);
+      setTimeout(function () {
+        location.reload();
+      }, 0);
+    } else if (
+      this.gameState[2] === "O" &&
+      this.gameState[5] === "O" &&
+      this.gameState[8] === "O"
+    ) {
+      setTimeout(function () {
+        alert("Player O has won!");
+      }, 0);
+      setTimeout(function () {
+        location.reload();
+      }, 0);
+    } else if (
+      this.gameState[0] === "O" &&
+      this.gameState[4] === "O" &&
+      this.gameState[8] === "O"
+    ) {
+      setTimeout(function () {
+        alert("Player O has won!");
+      }, 0);
+      setTimeout(function () {
+        location.reload();
+      }, 0);
+    } else if (
+      this.gameState[2] === "O" &&
+      this.gameState[4] === "O" &&
+      this.gameState[6] === "O"
+    ) {
+      setTimeout(function () {
+        alert("Player O has won!");
+      }, 0);
+      setTimeout(function () {
+        location.reload();
+      }, 0);
+    } else if (this.gameState.includes("") === false) {
+      setTimeout(function () {
+        alert("The game is draw!");
+      }, 0);
+      setTimeout(function () {
+        location.reload();
```

Comment:
This is one way to go about doing this and it works correctly. If you wanted to refactor this and save yourself a lot of lines of code, you could create an array of the winning combinations and iterate over the array to see if the content of those arrays all equal the same thing, then it would run your end game functions and alerts